tap-agent/src/agent/sender_account.rs

```diff
@@ -214,17 +287,32 @@ impl Actor for SenderAccount {
         );
         match message {
             SenderAccountMessage::UpdateReceiptFees(allocation_id, unaggregated_fees) => {
-                let tracker = &mut state.sender_fee_tracker;
-                tracker.update(allocation_id, unaggregated_fees.value);
+                state
+                    .sender_fee_tracker
+                    .update(allocation_id, unaggregated_fees.value);
 
-                if tracker.get_total_fee() >= state.config.tap.rav_request_trigger_value.into() {
+                // Eagerly deny the sender (if needed), before the RAV request. To be sure not to
+                // delay the denial because of the RAV request, which could take some time.
+                if !state.deny {
+                    state.maybe_deny().await;
+                }
+
+                if state.sender_fee_tracker.get_total_fee()
+                    >= state.config.tap.rav_request_trigger_value.into()
+                {
                     tracing::debug!(
-                        total_fee = tracker.get_total_fee(),
+                        total_fee = state.sender_fee_tracker.get_total_fee(),
                         trigger_value = state.config.tap.rav_request_trigger_value,
                         "Total fee greater than the trigger value. Triggering RAV request"
                     );
                     state.rav_requester_single().await?;
```

Comment:
You probably need to handle this, otherwise when the `rav_requester_single()` fails, the actor will be on a failed state.

Reply:
That's not code that I changed here though. Perhaps for another PR?

Reply:
sure